### ISSUE TITLE
Docker build improvements

### ISSUE THREAD

**Issue by aaronleopold** [aaronleopold/stump#27]
Creating separate issue off of #21. 

Currently, building the multi-arch docker images takes over 2 hours, mainly due to the arm platforms (the amd64 takes about 15 minutes). This is awful, and will pretty much prevent me implementing any sort of automated CI docker builds for Stump. Running the builds on my M1 laptop greatly reduces the build time for those architectures (arm64, arm/v7), but then the amd64 build fails from what I believe is related to qemu bug, based on my searches, but am unsure.

This issue is to, in general, overhaul the docker setup to allow for a CI pipeline to eventually be added once Stump is ready:

- [ ] Drastically reduce build times, as I can't pay for the CI minutes required for the current multi-hour docker build
- [ ] Support the following platforms:
    - [ ] arm64
    - [ ] arm/v7
    - [ ] amd64

**Comment by aaronleopold** [comment]
Closing in favor of #93
